File: backend/src/domain/models/transaction.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Transaction struct {
	ID        string `gorm:"primarykey"`
	CreatedAt time.Time
	UpdatedAt time.Time
	DeletedAt gorm.DeletedAt `gorm:"index"`

	TransactionTypeId string
	WalletId          string
	Name              string
	Description       string
	Value             float64
	TransactionDate   time.Time
}

File: src/domain/models/transaction.ts
interface TransactionWallet {
  id: string;
  name: string;
  description: string;
  created_at: string;
  updated_at: string;
}

interface TransactionCategory {
  id: string;
  name: string;
}

export interface Transaction {
  id?: string;
  name: string;
  description: string;
  value: string;
  transaction_date: string;
  wallet: TransactionWallet;
  categories: TransactionCategory[];
}
